### PR TITLE
iox-#1196 Fix clang-tidy warning in unique_id

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1196

### CODE REVIEW

**Comment by mossmaurice** [comment]
@elfenpiff Please add the reference to the original issue at the bottom, thanks!

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp [hunk 1]
```diff
@@ -45,7 +45,10 @@ class UniqueId : public cxx::NewType<uint64_t,
     UniqueId() noexcept;
 
   private:
-    static std::atomic<value_type> g_IdCounter; // initialized in corresponding cpp file
+    /// @NOLINTJUSTIFICATION only accessible by this class. the global variable is required to
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp:48]: AFAIK the `@` can be removed. I talked about this with @elBoberido recently. Same for all other suppressions.

Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp:48]: @elfenpiff Oh and could you maybe document the suppression syntax for clang tidy in `CONTRIBUTING.md`?

Comment (elfenpiff) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp:48]: I added it

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -133,6 +133,31 @@ information about logging and error handling.
...
+
+Furthermore, `NOLINTBEGIN` should only be used for a range as small as possible
+and maybe for a whole function but not more.
+
+#### Examples
+
+```cpp
+/// @NOLINTJUSTIFICATION we require the 'construct' to implement XXX and the safe usage
+///                         is guaranteed through YYY
+/// @NOLINTBEGIN(some-warning)
```
Comment (mossmaurice) [inline CONTRIBUTING.md:157]: @elfenpiff As mentioned [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1495#discussion_r920005056) the `@` can be omitted.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]
LTGM! It's ok for me to remove the `@` in the suppressions later. @elfenpiff Can you create a todo somewhere that we remove the `@` in a batch after clang-tidy is active everywhere?

**Pull request merged**